runtime/src/main/java/org/corfudb/runtime/clients/LogUnitSenderClient.java

```diff
@@ -0,0 +1,323 @@
+        });
+    }
+
+    /**
+     * Get the global tail maximum address the log unit has written.
+     *
+     * @return A CompletableFuture which will complete with the globalTail once
+     * received.
+     */
+    public CompletableFuture<Long> getTail() {
+        return router
+                .sendMessageAndGetCompletable(CorfuMsgType.TAIL_REQUEST.msg().setEpoch(epoch));
+    }
+
+    /**
+     * Get the starting address of a loggining unit.
+     *
+     * @return A CompletableFuture for the starting address
+     */
+    public CompletableFuture<Long> getTrimMark() {
+        return router
+                .sendMessageAndGetCompletable(CorfuMsgType.TRIM_MARK_REQUEST.msg().setEpoch(epoch));
+    }
+
+    /**
+     * Send a hint to the logging unit that a stream can be trimmed.
+     *
+     * @param prefix The prefix of the stream, as a global physical offset, to trim.
+     */
+    public void trim(long prefix) {
+        router.sendMessage(CorfuMsgType.TRIM.payloadMsg(
+                new TrimRequest(null, prefix)).setEpoch(epoch));
+    }
+
+    /**
+     * Send a prefix trim request that will trim the log up to a certian address
+     *
+     * @param address An address to trim up to (i.e. [0, address))
+     */
+    public CompletableFuture<Void> prefixTrim(long address) {
+        return router.sendMessageAndGetCompletable(CorfuMsgType.PREFIX_TRIM
+                .payloadMsg(new TrimRequest(null, address)).setEpoch(epoch));
+    }
+
+    /**
+     * Send a compact request that will delete the trimmed parts of the log.
+     */
+    public CompletableFuture<Void> compact() {
+        return router
+                .sendMessageAndGetCompletable(CorfuMsgType.COMPACT_REQUEST.msg().setEpoch(epoch));
+    }
+
+    /**
+     * Send a flush cache request that will flush the logunit cache.
+     */
+    public CompletableFuture<Void> flushCache() {
+        return router
+                .sendMessageAndGetCompletable(CorfuMsgType.FLUSH_CACHE.msg().setEpoch(epoch));
+    }
+
+    /**
+     * Fill a hole at a given address.
+     *
+     * @param address The address to fill a hole at.
+     */
+    public CompletableFuture<Boolean> fillHole(long address) {
+        Timer.Context context = getTimerContext("fillHole");
+        CompletableFuture<Boolean> cf = router.sendMessageAndGetCompletable(
+                CorfuMsgType.FILL_HOLE.payloadMsg(
+                        new FillHoleRequest(null, address)).setEpoch(epoch));
+        return cf.thenApply(x -> {
+            context.stop();
+            return x;
+        });
+    }
+
+    /**
+     * Fills hole at a given address for a particular streamID.
+     *
+     * @param streamID StreamID to hole fill.
+     * @param address  The address to fill a hole at.
+     */
+    @Deprecated // TODO: Add replacement method that conforms to style
+    @SuppressWarnings("checkstyle:abbreviation") // Due to deprecation
+    public CompletableFuture<Boolean> fillHole(UUID streamID, long address) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)